experimental/cuda2/cuda_driver.c

```diff
@@ -0,0 +1,448 @@
+  (void)driver;
+  (void)device;
+  return iree_ok_status();
+}
+
+static iree_status_t iree_hal_cuda2_driver_select_default_device(
+    iree_hal_driver_t* base_driver, iree_hal_cuda2_dynamic_symbols_t* syms,
+    int default_device_index, iree_allocator_t host_allocator,
+    CUdevice* out_device) {
+  iree_hal_device_info_t* device_infos = NULL;
+  iree_host_size_t device_count = 0;
+  IREE_RETURN_IF_ERROR(iree_hal_cuda2_driver_query_available_devices(
+      base_driver, host_allocator, &device_count, &device_infos));
+
+  iree_status_t status = iree_ok_status();
+  if (device_count == 0) {
+    status = iree_make_status(IREE_STATUS_UNAVAILABLE,
+                              "no compatible CUDA devices were found");
+  } else if (default_device_index >= device_count) {
+    status = iree_make_status(IREE_STATUS_NOT_FOUND,
+                              "default device %d not found (of %ld enumerated)",
+                              default_device_index, device_count);
+  } else {
+    *out_device =
+        DEVICE_ID_TO_CUDEVICE(device_infos[default_device_index].device_id);
+  }
+  iree_allocator_free(host_allocator, device_infos);
+
+  return status;
+}
+
+static iree_status_t iree_hal_cuda2_driver_create_device_by_id(
+    iree_hal_driver_t* base_driver, iree_hal_device_id_t device_id,
+    iree_host_size_t param_count, const iree_string_pair_t* params,
+    iree_allocator_t host_allocator, iree_hal_device_t** out_device) {
+  IREE_ASSERT_ARGUMENT(base_driver);
+  IREE_ASSERT_ARGUMENT(params);
+  IREE_ASSERT_ARGUMENT(out_device);
+
+  iree_hal_cuda2_driver_t* driver = iree_hal_cuda2_driver_cast(base_driver);
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  // Ensure cuda2 is initialized before querying it.
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(z0, iree_hal_cuda2_init(driver));
+
+  // Use either the specified device (enumerated earlier) or whatever default
+  // one was specified when the driver was created.
+  CUdevice device = 0;
+  if (device_id == IREE_HAL_DEVICE_ID_DEFAULT) {
+    IREE_RETURN_AND_END_ZONE_IF_ERROR(
+        z0, iree_hal_cuda2_driver_select_default_device(
+                base_driver, &driver->cuda_symbols,
+                driver->default_device_index, host_allocator, &device));
+  } else {
+    device = DEVICE_ID_TO_CUDEVICE(device_id);
+  }
+  (void)device;
+
+  IREE_TRACE_ZONE_END(z0);
+  return iree_status_from_code(IREE_STATUS_UNIMPLEMENTED);
+}
+
+static iree_status_t iree_hal_cuda2_driver_create_device_by_uuid(
+    iree_hal_driver_t* base_driver, iree_string_view_t driver_name,
+    const CUuuid* device_uuid, iree_host_size_t param_count,
+    const iree_string_pair_t* params, iree_allocator_t host_allocator,
+    iree_hal_device_t** out_device) {
+  iree_hal_cuda2_driver_t* driver = iree_hal_cuda2_driver_cast(base_driver);
+
+  // Ensure cuda2 is initialized before querying it.
+  IREE_RETURN_IF_ERROR(iree_hal_cuda2_init(driver));
+
+  // cuda2 doesn't have an API to do this so we need to scan all devices to
```

Comment:
```suggestion
  // CUDA doesn't have an API to do this so we need to scan all devices to
```
(maybe find/replace was too aggressive - use `CUDA` when referring to the API)
here and elsewhere in these files

Reply:
Yup! Too aggressive find and replace. :) I reverted not suitable places.

Reply:
Also checked that we don't have such cases anymore in other files.